index.js

```diff
@@ -161,36 +199,26 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
       Metadata: opts.metadata,
       StorageClass: opts.storageClass,
       ServerSideEncryption: opts.serverSideEncryption,
-      SSEKMSKeyId: opts.sseKmsKeyId,
-      Body: (opts.replacementStream || file.stream)
+      SSEKMSKeyId: opts.sseKmsKeyId
     }
 
     if (opts.contentDisposition) {
       params.ContentDisposition = opts.contentDisposition
     }
 
-    var upload = this.s3.upload(params)
-
-    upload.on('httpUploadProgress', function (ev) {
-      if (ev.total) currentSize = ev.total
-    })
-
-    upload.send(function (err, result) {
-      if (err) return cb(err)
+    if (file.mimetype !== 'image/svg+xml') {
+      params.Body = stream
+      var upload = this.s3.upload(params)
+      return initializeUploadEvents(upload, opts, cb)
+    }
 
-      cb(null, {
-        size: currentSize,
-        bucket: opts.bucket,
-        key: opts.key,
-        acl: opts.acl,
-        contentType: opts.contentType,
-        contentDisposition: opts.contentDisposition,
-        storageClass: opts.storageClass,
-        serverSideEncryption: opts.serverSideEncryption,
-        metadata: opts.metadata,
-        location: result.Location,
-        etag: result.ETag
-      })
+    file.buffer = ''
```

Comment:
Seems like if we use an npm module to read this in (which i think it all-around the better idea), we won't be able to check file size as we go. We can check after reading in the file, but before sending it to s3? I'm not sure exactly what the filesize check is trying to prevent, so if that's sufficient, I can do the check before s3 upload. (Also, what is the intended limit on our filesize?)